Wassaf/28 ocw hugo footer

Wassaf-Shahzad opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #28  

#### What's this PR do?

- Updates footer as per invision design 
- Adds footer to Home Page About and Educator Page

#### How should this be manually tested?
- Verify footer is responsive
- Verify all links point to correct URLs
- Verify spacing and font sizes

#### Screenshots (if appropriate)

![Screenshot 2022-01-14 at 3 15 59 AM](https://user-images.githubusercontent.com/87968618/149423412-8dcbb034-9960-4325-bdde-4d104a62333c.png)

![Screenshot 2022-01-14 at 3 15 52 AM](https://user-images.githubusercontent.com/87968618/149423617-f3d00bf0-fc31-4b92-8ac1-c8fc5156c681.png)

![Screenshot 2022-01-14 at 4 16 19 AM](https://user-images.githubusercontent.com/87968618/149423963-cefe1b35-0441-4ea9-9d43-487ea5fada05.png)

![Screenshot 2022-01-14 at 4 16 47 AM](https://user-images.githubusercontent.com/87968618/149423977-66cbb62b-1b74-4197-a366-47a301c4339a.png)

![Screenshot 2022-01-14 at 4 17 01 AM](https://user-images.githubusercontent.com/87968618/149423990-b4f3fa56-3b44-46e4-a5c2-5b8d32787a95.png)

<img width="1424" alt="Screenshot 2022-01-26 at 4 44 59 PM" src="https://user-images.githubusercontent.com/87968618/151157831-9f50db35-9b25-40b9-8eeb-c37f523a53fe.png">

Wassaf-Shahzad commented:
@pdpinch.
I currently just need the icon images for the social websites after which this PR will be ready to go.

MAbdurrehman12 reviewed (changes requested):
Footer looks great! just noticed few design things (did not review the code yet, it can be reviewed after this I guess):
 

1. I guess all the "non-ocw" links, such as links to mit, openlearning, especially social media, should be opened in a new tab? Perhaps @pdpinch can share his thoughts.
2. Not sure, but what if we keep the social media links in their original color instead of all being blue? (also dropped a comment in design)
3. Shouldn't the copyright text include 2022 instead of 2021? (also dropped a comment in design)
4. For large screens, having width from 992px to somewhere around 1190px, the social media links section comes close to the first section, instead of being float/pushed to right. (Not sure, you can also verify). For reference please see **image#1**.
5. For mobile screens, having width 375px, the 2 logos seems to be overlapping. I guess these logos can be aligned vertically (instead of being aligned horizontally) for this screen size. Please see **image#2**
6. Same issue as above, open education global, can be aligned vertically with it's heading. 
Please see **image#3** 

Reference images:

**image#1**
![image](https://user-images.githubusercontent.com/93309234/149491636-6f58b907-53d3-4edc-876f-b29fae739dc3.png)

**image#2**
![image](https://user-images.githubusercontent.com/93309234/149492071-509f6499-91a1-4c6c-a60e-4a883cb91799.png)

**image#3**
![image](https://user-images.githubusercontent.com/93309234/149492275-2cb747fd-21fa-49a9-bacd-fd18546adbf3.png)

MAbdurrehman12 commented on base-theme/assets/css/footer.scss:
```diff
@@ -0,0 +1,130 @@
+  padding-right: 40px;
+  justify-content: space-between;
+
+  .abc {
```
I wonder if this is a correct name for any class and is it being used somewhere?

MAbdurrehman12 commented on base-theme/layouts/partials/footer.html:
```diff
@@ -1,2 +1,53 @@
-<footer>
+<footer id="footer_container" class="fluid_container column-below-medium">
```
is "fluid_container" a customized class? As I couldn't find it.
Maybe we can use bootstrap class "container-fluid" (if required)

I'm also not sure about the naming conventions of CSS classes but I guess it's all lowercase separated with '-'

MAbdurrehman12 commented on base-theme/layouts/partials/footer.html:
```diff
@@ -1,2 +1,53 @@
+      </div>
+      <div class="horizontal_list">
+        <ul>
+          <li><a href="https://www.facebook.com/MITOCW/"><i class="material-icons blue600">facebook</i></a></li>
```
just wondering if "blue600" is a correct convention for a CSS class. 
ps: Is it a customized class? As I couldn't find it

Wassaf-Shahzad commented:
> Footer looks great! just noticed few design things:
> 
> 1. I guess all the "non-ocw" links, such as links to mit, openlearning, especially social media, should be opened in a new tab? Perhaps @pdpinch can share his thoughts.
> 2. Not sure, but what if we keep the social media links in their original color instead of all being blue? (also dropped a comment in design)
> 3. Shouldn't the copyright text include 2022 instead of 2021? (also dropped a comment in design)
> 4. For large screens, having width from 992px to somewhere around 1190px, the social media links section comes close to the first section, instead of being float/pushed to right. (Not sure, you can also verify). For reference please see **image#1**.
> 5. For mobile screens, having width 375px, the 2 logos seems to be overlapping. I guess these logos can be aligned vertically (instead of being aligned horizontally) for this screen size. Please see **image#2**
> 6. Same issue as above, open education global, can be aligned vertically with it's heading.
>    Please see **image#3**
> 
> Reference images:
> 
> **image#1** ![image](https://user-images.githubusercontent.com/93309234/149491636-6f58b907-53d3-4edc-876f-b29fae739dc3.png)
> 
> **image#2** ![image](https://user-images.githubusercontent.com/93309234/149492071-509f6499-91a1-4c6c-a60e-4a883cb91799.png)
> 
> **image#3** ![image](https://user-images.githubusercontent.com/93309234/149492275-2cb747fd-21fa-49a9-bacd-fd18546adbf3.png)

I have addressed all the feedback However I do want to inform that Open learning Image will still be unaligned at  < 400px
since the image itself has right padding I will add comments in the invision design to request that image

<img width="500" alt="Screenshot 2022-01-17 at 3 10 10 PM" src="https://user-images.githubusercontent.com/87968618/149755198-cf7c215f-93fc-4bde-8d68-880c0b7cae9b.png">

<img width="500" alt="Screenshot 2022-01-17 at 3 09 48 PM" src="https://user-images.githubusercontent.com/87968618/149755237-d982db26-cc07-40df-b5e9-e3276d1a86f4.png">

MAbdurrehman12 reviewed (approved):
👍

pdpinch commented:
Sorry I missed these questions. 


1. Yes please open the off-site links in a new window.  
2. Did you get a reply from the designer? If not, just leave them as they are now. 
3. Yes copyright should be 2022
4. I think this is OK. 
5. Yes, please try aligning vertically
6. This doesn't look too bad to me, but you can try aligning vertically if you like.

pdpinch commented:
please don't merge this yet. The MIT logo, and the social network logos need attention, but I need to talk to some other folks to figure out what to do with them.

Wassaf-Shahzad commented:
> please don't merge this yet. The MIT logo, and the social network logos need attention, but I need to talk to some other folks to figure out what to do with them.

@pdpinch Sure I will wait.
However I had added comments on the invision design but received no reply from the designer and I am aware of the social links issue but not of the MIT logo.

Wassaf-Shahzad commented:
@pdpinch @MAbdurrehman12 Updated social images

pdpinch commented:
Sorry @Wassaf-Shahzad you questions brought up some issues with the footer design. Can you update to use his new design? I hope it's not a lot of work. https://invis.io/UM12AT5562GS#/463497214_OCW_Footer_2022_V1

pdpinch commented on base-theme/layouts/partials/footer.html:
```diff
@@ -1,2 +1,62 @@
+    <section class="columns">
+      <div class="pd-14">
+        <ul>
+          <li><a href="https://accessibility.mit.edu">ACCESSIBILITY</a></li>
```
Why is this in all caps?

```suggestion
          <li><a href="https://accessibility.mit.edu">Accessibility</a></li>
```

pdpinch commented on base-theme/layouts/partials/footer.html:
```diff
@@ -1,2 +1,62 @@
+        <ul>
+          <li><a href="https://accessibility.mit.edu">ACCESSIBILITY</a></li>
+          <li><a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons License</a></li>
+          <li><a href="https://ocw.mit.edu/terms/">Terms and Conditions</a></li>
```
I'm looking to see if we have another URL for this page.

pdpinch replied:
```suggestion
          <li><a href="pages/privacy-and-terms-of-use/">Terms and Conditions</a></li>
```

I'd like to use a page that managed by the CMS. I think this will be the correct URL.

pdpinch commented on base-theme/layouts/partials/footer.html:
```diff
@@ -1,2 +1,62 @@
+          </a></li>
+          <li> <a class="img-link" href="https://www.youtube.com/mitocw" target="_blank">
+            <img class="social-icon" src="/images/social-youtube-blue.svg" alt="youtube" />
+          </a>
```
Add a logo and link for LinkedIn as well. It should go to https://www.linkedin.com/company/mit-opencourseware/

pdpinch commented on base-theme/layouts/partials/footer.html:
```diff
@@ -1,2 +1,62 @@
+        </ul>
+      </div>
+      <div class="copyright large-and-above-only">
+        <p>© 2001–2021 	Massachusetts Institute of Technology</p>
```
```suggestion
        <p>© 2001–2022 	Massachusetts Institute of Technology</p>
```

MAbdurrehman12 reviewed (changes requested):
Nice work 👍 

1 thing which I just noticed that for `Educator Page`, a red border was added for mobile view in this [PR](https://github.com/mitodl/ocw-hugo-themes/pull/332/), the css was already provided for it and I have no idea why the border was made red only for the educator page. 
Now I have 2 issues in mind: 
1- That red border is visible above footer as shown in ref. image# 1  
2- It looks like only educator page has this border which makes it inconsistent with the border of other pages.

@pdpinch Can you also share your thoughts on this please?

ref. image# 1:
![image](https://user-images.githubusercontent.com/93309234/151935595-2b294aeb-0a42-438b-bcc7-336a6b65e035.png)

pdpinch commented:
Sorry, I have no idea what that red border is for. Let's discuss tomorrow, but it shouldn't block this PR.

pdpinch commented on base-theme/layouts/partials/footer.html:
```diff
@@ -1,2 +1,66 @@
+          <ul>
+            <li><a href="https://accessibility.mit.edu">Accessibility</a></li>
+            <li><a href="https://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons License</a></li>
+            <li><a href="https://ocw.mit.edu/terms/">Terms and Conditions</a></li>
```
```suggestion
            <li><a href="pages/privacy-and-terms-of-use/">Terms and Conditions</a></li>
```

I'd like to link this to a page managed by OCW Studio. I think this URL will work.

Wassaf-Shahzad replied:
I checked RC and its giving me a 404 for this URL ie https://ocwnext-rc.odl.mit.edu/pages/privacy-and-terms-of-use/

pdpinch replied:
It's working now. I think the page needed to be published.

pdpinch reviewed (changes requested):
Does this PR add the footer to the course theme as well, or will that be in another PR?

Wassaf-Shahzad commented:
> Does this PR add the footer to the course theme as well, or will that be in another PR?

This PR does not add the footer in the course theme however the footer was already added in the initial commit for the course theme .

Wassaf-Shahzad commented:
@pdpinch addressed changes

MAbdurrehman12 reviewed (approved):
👍
